src/controllers/controller.doughnut.js

```diff
@@ -276,18 +276,41 @@ module.exports = DatasetController.extend({
 
 	// gets the max border or hover width to properly scale pie charts
 	getMaxBorderWidth: function(arcs) {
+		var me = this;
 		var max = 0;
-		var index = this.index;
-		var length = arcs.length;
-		var borderWidth;
-		var hoverWidth;
+		var index = me.index;
+		var chart = me.chart;
+		var valueAtIndexOrDefault = helpers.valueAtIndexOrDefault;
+		var elementOpts = chart.options.elements.arc;
+		var i, ilen, dataset, custom, borderAlign, borderWidth, hoverWidth;
+
+		if (!arcs) {
+			// Find the outmost visible dataset
+			for (i = 0, ilen = chart.data.datasets.length; i < ilen; ++i) {
+				if (chart.isDatasetVisible(i)) {
+					arcs = chart.getDatasetMeta(i).data;
+					index = i;
+					break;
+				}
+			}
+		}
+
+		if (!arcs) {
+			return 0;
+		}
 
-		for (var i = 0; i < length; i++) {
-			borderWidth = arcs[i]._model ? arcs[i]._model.borderWidth : 0;
-			hoverWidth = arcs[i]._chart ? arcs[i]._chart.config.data.datasets[index].hoverBorderWidth : 0;
+		dataset = chart.data.datasets[index];
 
-			max = borderWidth > max ? borderWidth : max;
-			max = hoverWidth > max ? hoverWidth : max;
+		for (i = 0, ilen = arcs.length; i < ilen; ++i) {
+			custom = arcs[i].custom || {};
+			borderAlign = custom.borderAlign ? custom.borderAlign : valueAtIndexOrDefault(dataset.borderAlign, i, elementOpts.borderAlign);
```

Comment:
`_resolveElementOptions` has been added. As the visible outmost dataset can be different from the current one, the cached values is used only when they are the for the current dataset in `getMaxBorderWidth`.

Reply:
I didn't realize that `getMaxBorderWidth()`, when called without parameters, was computing the width based on the outmost visible dataset, which is most of the time not the same as the current one when there is more than 1 dataset.

It's not really optimal because `_resolveElementOptions` get called `2 * datasets.length - 1` times per `chart.update()` while ideally `getMaxBorderWidth()` should be called only one time, so  `_resolveElementOptions` only `datasets.length` times. This can be a performance issue when we will introduce scriptable options but not sure how to make that better right now since there is no way to centralize these computations for all datasets.